slug.go

```diff
@@ -34,14 +34,65 @@ func (e *IllegalSlugError) Error() string {
 // chain.
 func (e *IllegalSlugError) Unwrap() error { return e.Err }
 
+// PackerOption is a functional option that can configure non-default Packers.
+type PackerOption func(*Packer) error
+
+// BypassIgnore is a PackerOption that will skip the .terraformignore rules and
+// Pack the entire src directory.
+func BypassIgnore() PackerOption {
```

Comment:
Instead of having an option to _disable_ the `.terraformignore` file, we should consider having an option to _enable_ it. This package is used in many contexts that are not Terraform, an as such I think that the default should be as unsurprising as possible.

We can still maintain the package-level `Pack()` and have that default to applying the `.tfignore`, such that we are fully backwards compatible and do not introduce any regressions. But I'd strive for `NewPacker().Pack()` to default to packaging in a generic way without any fancy .terraformignore logic.

Reply:
This makes a lot of sense to me. It fits well with the "make the zero value useful" rule, and it fixes my awkward "if not not applying" if statement. And like you said, since we can set whatever rules in the package level `Pack` we want, it can be invisible to current callers. Will do.